### PR TITLE
ENG-15071: A program for Calcite planning comparison	

### PR DESCRIPTION
**Pull request by Chaozz**
A command line tool for Calcite planning comparison. You can use the script at `tests/sqlgrammar/plan-checker` to run it.
```
usage: plan-checker
 -d,--ddl <arg>     File path for the DDLs.
 -f,--file <arg>    The input file path that contains SQL query statements
                    (If -q/--query is used, this option will be ignored).
 -q,--query <arg>   Single SQL query statement string.
```
If the query is planable and there is any difference between the orignal volt plan and the new calcite plan, it will print out the query together with the two plans in **STDERR**. Otherwise, it will print nothing in STDERR. For example:
```console
foo@bar:~$ plan-checker -d testcalcite-ddl.sql -q "select i * 5 from R1"
select i * 5 from R1
Expected: {"PLAN_NODES":[{"ID":1,"PLAN_NODE_TYPE":"SEND","CHILDREN_IDS":[2]},{"ID":2,"PLAN_NODE_TYPE":"SEQSCAN","INLINE_NODES":[{"ID":3,"PLAN_NODE_TYPE":"PROJECTION","OUTPUT_SCHEMA":[{"COLUMN_NAME":"C1","EXPRESSION":{"TYPE":3,"VALUE_TYPE":6,"LEFT":{"TYPE":32,"VALUE_TYPE":5,"COLUMN_IDX":0},"RIGHT":{"TYPE":30,"VALUE_TYPE":5,"ISNULL":false,"VALUE":5}}}]}],"TARGET_TABLE_NAME":"R1","TARGET_TABLE_ALIAS":"R1"}]}
Actual: {"PLAN_NODES":[{"ID":1,"PLAN_NODE_TYPE":"SEND","CHILDREN_IDS":[2]},{"ID":2,"PLAN_NODE_TYPE":"SEQSCAN","INLINE_NODES":[{"ID":3,"PLAN_NODE_TYPE":"PROJECTION","OUTPUT_SCHEMA":[{"COLUMN_NAME":"C1","EXPRESSION":{"TYPE":3,"VALUE_TYPE":5,"LEFT":{"TYPE":32,"VALUE_TYPE":5,"COLUMN_IDX":0},"RIGHT":{"TYPE":30,"VALUE_TYPE":5,"ISNULL":false,"VALUE":5}}}]}],"TARGET_TABLE_NAME":"R1","TARGET_TABLE_ALIAS":"R1"}]}
```

### CODE REVIEW

File: tests/sqlgrammar/plan-checker [hunk 1]
```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env bash
+
+# This file is part of VoltDB.
+
+# Copyright (C) 2008-2019 VoltDB Inc.
```
Comment (johnTcrawford) [inline tests/sqlgrammar/plan-checker]: When I write a bash script, I usually use the .sh file extension, and do not include the standard VoltDB Copyright/license message (which licensecheck does not mind, unlike, e.g., .java files); however, I'm not sure whether those two habits are our standard policy, or just what I do.

File: tests/frontend/org/voltdb/plannerv2/PlanChecker.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        } catch (Exception e) {
+            // The SQL is invalid in VOLT, stop checking and print nothing
+            return;
+        }
+        try {
+            voltdbPlan = compileAdHocPlan(sql, true, true);
+        } catch (Exception e) {
+            // The SQL is not supported in Calcite planner, stop checking and print nothing
+            return;
+        }
```
[the remark this answers is not in the document]
Comment (Chaozz) [reply tests/frontend/org/voltdb/plannerv2/PlanChecker.java]: good catch. Done.

File: tests/frontend/org/voltdb/plannerv2/PlanChecker.java [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        try {
+            if (!CALCITE_CHECKS.check(sql)) {
+                // The query cannot pass the compatibility check, we just ignore it.
+                return;
+            }
+        } catch (SqlParseException e) {
+            return;
+        }
```
[the remark this answers is not in the document]
Comment (Chaozz) [reply tests/frontend/org/voltdb/plannerv2/PlanChecker.java]: done

File: tests/sqlgrammar/plan-checker (hunk 1 above)
Comment (Chaozz) [reply tests/sqlgrammar/plan-checker]: I follow the pattern of how the scripts are formatted in the `voltdb/bin` folder. I think it's not a big deal so I just keep as it is.

**Pull request merged**